### ISSUE TITLE
Pretitle as heading

### ISSUE THREAD

**Issue by aweell** [Telefonica/mistica-design#1818]
## Description

There are some use case where is needed to tag the pretitle element as a heading instead of the title.

The affected elements are:

* Header
* Hero & Cover hero
* Cards
* Empty-state
* Callout

The specifications must reflect that it is possible to change the tag of the pretitle an the title, and the logic described in the notes section of this issue. 

### Notes

* How this impact the current reading order of the cards? Should we change the order depending if the pretitle or the title are the heading? What happen if there's a use case both of them are? I should be the higher heading first?.

SOLVED: 

* If there's only one heading, the heading will be read first
* If both pretitle and title are a heading, the heading with the higher hierarchy will be read first
* Pretitle and title cannot have the same level of heading

## Tasks

- [x] Update specs
   - [x] [Header specs](https://www.figma.com/design/Xm3r8nRNlbAiZNlkaH5YX1/branch/IMfDTrMkkFR3YD3tyMDs2h/%F0%9F%94%B8-Header-Component-Specs?m=auto&node-id=1875-7013&t=QWOd3NiskFLp0cu3-1)
   - [x] [Hero & cover hero specs](https://www.figma.com/design/M8t65SWdTUyM44YjxuzCyu/branch/kwfW3gTeU3xl0RwuFbQ0Hq/%F0%9F%94%B8-Hero-Specs?m=auto&node-id=3761-17008&t=A4MgqSvmg0TDMf2N-1)
   - [x] [Cards specs](https://www.figma.com/design/tKdPOfcUALzVIh5oizFbm7/branch/9Zs9SdrcDMdnCmCVxRfSjY/%F0%9F%94%B8-Cards-Specs?m=auto&node-id=9507-6893&t=V90930T8zmZXLaiS-1)
   - [x] Empty-state specs - No aplica, no tiene pretitle
   - [x] Callout - No aplica, no tiene pretitle
- [ ] Design documentation
- [x] [Web ticket](https://github.com/Telefonica/mistica-web/pull/1170)
- [x] [Web ticket (orden lectura title & pretitle)](https://jira.tid.es/browse/WEB-2087)

**Comment by yceballost** [comment]
Pending to be done by @MontseTelefonica 

https://jira.tid.es/browse/PRODUCTDSN-2500
